tests/UrlTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/*
+ * This file is part of the webmozart/path-util package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Webmozart\PathUtil\Tests;
+
+use Webmozart\PathUtil\Url;
+
+/**
+ * @since  2.2
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Claudio Zizza <claudio@budgegeria.de>
+ */
+class UrlTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider provideMakeRelativeTests
+     * @covers Webmozart\PathUtil\Url
+     */
+    public function testMakeRelative($absoluteUrl, $baseUrl, $relativePath)
+    {
+        $relative = Url::makeRelative($absoluteUrl, $baseUrl);
+        $this->assertSame($relativePath, $relative);
+    }
+
+    /**
+     * @dataProvider provideMakeRelativeTests
+     * @covers Webmozart\PathUtil\Url
+     */
+    public function testMakeRelativeWithUrl($absoluteUrl, $baseUrl, $relativePath)
+    {
+        $relative = Url::makeRelative($this->createUrl($absoluteUrl), $this->createUrl($baseUrl));
+        $this->assertSame($relativePath, $relative);
+    }
+
+    /**
+     * @dataProvider provideMakeRelativeTests
+     * @covers Webmozart\PathUtil\Url
+     */
+    public function testMakeRelativeWithFullUrl($absoluteUrl, $baseUrl, $relativePath)
+    {
+        $url = 'ftp://user:password@example.com:8080';
+
+        $relative = Url::makeRelative($this->createUrl($absoluteUrl, $url), $this->createUrl($baseUrl, $url));
+        $this->assertSame($relativePath, $relative);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The path must be a string. Got: array
+     * @covers Webmozart\PathUtil\Url
+     */
+    public function testMakeRelativeFailsIfInvalidPath()
+    {
+        Url::makeRelative(array(), 'http://example.com/webmozart/puli');
+    }
+
```

Comment:
IMO you should also add `testMakeRelativeFailsIfNoUrl` here that tests that the following fails:

```php
Url::makeRelative('/webmozart/puli/css/style.css', '/webmozart/puli');
```

The `Url` class should only deal with URLs, not with paths.